src/kernel/context.cpp

```diff
@@ -5,25 +5,35 @@
 #include <kernel/context.h>
 
 #include <crypto/sha256.h>
+#include <kernel/checks.h>
 #include <key.h>
 #include <logging.h>
 #include <pubkey.h>
 #include <random.h>
+#include <util/result.h>
+#include <util/time.h>
 
 #include <string>
 
-
 namespace kernel {
 Context* g_context;
 
-Context::Context()
+util::Result<std::unique_ptr<Context>> Context::MakeContext()
 {
     assert(!g_context);
-    g_context = this;
+    auto context{std::unique_ptr<Context>(new Context())};
+    g_context = context.get();
     std::string sha256_algo = SHA256AutoDetect();
     LogPrintf("Using the '%s' SHA256 implementation\n", sha256_algo);
     RandomInit();
     ECC_Start();
+
+    if (auto res{SanityChecks(*context)}; !res) {
+        LogPrintf("Kernel sanity check failed: %s\n", ErrorString(res).original);
```

Comment:
Isn't this logged in the caller as well?

Reply:
Yeah, I'll remove this line.